src/settings.cpp

```diff
@@ -1585,20 +1668,32 @@ static IniFile *IniLoadConfig()
  */
 void LoadFromConfig(bool startup)
 {
-	IniFile *ini = IniLoadConfig();
+	IniFile *generic_ini = IniLoadConfig();
+	IniFile *private_ini = new IniFile();
```

Comment:
Most of IniLoadConfig is passing `_list_group_names` to the created IniFile. Looking at those names I see bans, servers and server_bind_addresses which go to the private ini. Shouldn't that constructor be called with the appropriate names for (at least) the private ini? Otherwise that won't get the type of bans set to IGT_LIST and I wouldn't know what kinds of repercussions that would have.
Those `list_group_names` do not seem to do much harm, so maybe a:
```
class ConfigIniFile : public IniFile {
public:
    ConfigIniFile(const std::string &filename) : IniFile(_list_group_names) 
    {
        this->LoadFromDisk(filename, NO_DIRECTORY);
    }
```
and then 
```
ConfigIniFile generic_ini(_config_file);
ConfigIniFile private_ini(_private_file);
ConfigIniFile secrets_ini(_secrets_file);
```
and get rid of the `delete` at the end of the function and of course of IniLoadConfig. Might also mean that you can pass them as reference to HandleSettingDesc, though that might be a bridge too far for this PR.

Reply:
I might have taken this a bit to an extreme, but it makes the code a whole lot better and more C++. So yeah, sounds like a good plan :D (well, I write this, but you can't see it yet, till I push the new version .. the suspense!)